backend/src/apiserver/main.go

```diff
@@ -189,6 +195,24 @@ func loadSamples(resourceManager *resource.ResourceManager) error {
 		if configErr != nil {
 			return fmt.Errorf("Failed to decompress the file %s. Error: %v", config.Name, configErr)
 		}
+		// Patch the default bucket name read from ConfigMap
+		if common.GetBoolConfigWithDefault(HasDefaultBucketEnvVar, false) {
+			defaultBucket := common.GetStringConfig(DefaultBucketNameEnvVar)
+			projectId := common.GetStringConfig(ProjectIDEnvVar)
+			patchMap := map[string]string{
+				"<your-gcs-bucket>": defaultBucket,
+				"<your-project-id>": projectId,
+			}
+			if config.Name == "[Sample] ML - XGBoost - Training with Confusion Matrix" {
```

Comment:
I think it might be fine to do the patching for all preloaded samples.

Reply:
Yes that makes sense. Changed to patch all samples.